app/src/main/kotlin/com/intive/picover/camera/view/CameraScreen.kt

```diff
@@ -16,13 +16,12 @@ import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
 import com.intive.picover.R
 import com.intive.picover.camera.viewmodel.CameraViewModel
-import com.intive.picover.common.loader.PicoverLoader
 import com.skydoves.landscapist.ImageOptions
 import com.skydoves.landscapist.coil.CoilImage
 
 @Composable
 fun CameraScreen(viewModel: CameraViewModel) {
-	val takenImageUri by viewModel.takenImageUri
+	val takenImageUri = viewModel.takenImageUri
 	val isImageTaken by viewModel.isImageTaken
 	val cameraLauncher = rememberLauncherForActivityResult(ActivityResultContracts.TakePicture()) {
 		viewModel.onImageTaken(wasSaved = it)
```

Comment:
`viewModel.onImageTaken` should be called only, as name says, when image was taken, not for every result from `TakePicture` contract. 
In case of canceled photo, contract returns `false`, which is propagated down to the view model but it shouldn't (parameter `wasSaved` - name is the implication of passed flag) - check `CameraScreen::onImageTaken`